### PR TITLE
feat: add default gtm integration

### PR DESCRIPTION
**Pull request by MrAvantiC**

### CODE REVIEW

**Review by sauerben — COMMENTED** [review]
@MrAvantiC 
Looks fine. And I think it will cover the basic needs for tracking page views. 
I have some points, but they are mostly personal opinion:
- I would prefer to don't have the `gmt.js` in the repo but in a dependency (like https://www.npmjs.com/package/react-gtm-module which is very small, was not updated in 5 years but still)
- I like the solution using hooks like we did in some projects. Because they come in handy for tracking clicks, impression and other things all using the same hooks.
- In my mind, tracking is a byproduct and so hooks are super handy because you can keep all the tracking logic out of your components.
- I don't like that the entry.js and the search.js are getting longer. In some projects when we split the pages we have even more duplicate code (maybe https://nextjs.org/blog/layouts-rfc will come in handy in the future). 
- Because the entry.js and search.js are Class Components, we cannot use hooks. Is there a plan to change it (and go all into functional components 😍)

**Comment by MrAvantiC** [comment]
Thanks for taking a look!

> * I would prefer to don't have the `gmt.js` in the repo but in a dependency (like https://www.npmjs.com/package/react-gtm-module which is very small, was not updated in 5 years but still)

Since the implementation is pretty simple so far, I would prefer to not rely on external dependencies. As you already noted, the package hasn't been updated in years. And just by looking at the docs, I could not see a way to configure the server the GTM-Script is loaded from ...which is quite important now that we want to implement Server-Side-Containers as well.

> * I like the solution using hooks like we did in some projects. Because they come in handy for tracking clicks, impression and other things all using the same hooks.

I'm not sure why we would limit us to using our functionality only within components. Because this is what we are doing when we implement everything inside hooks. What about tracking from within util-functions or similar?
Also, I'm not sure how it makes tracking clicks or impressions easier.
However, we will use this as a basis now to implement in a new project. So, I would keep this MR open for now and take the learnings from this project back here once everything is implemented. Maybe one of the learnings is that hooks are indeed more suitable. ;) 

> * In my mind, tracking is a byproduct and so hooks are super handy because you can keep all the tracking logic out of your components.

Like I said above, it feels the other way around to me. Hooks actually force use to stay within the components. I'm trying to avoid the same pain I sometimes have when I need translations and have to pass them through a component because - right now - they are only available using the `useTranslation` hook.

> * I don't like that the entry.js and the search.js are getting longer. In some projects when we split the pages we have even more duplicate code (maybe https://nextjs.org/blog/layouts-rfc will come in handy in the future).

Do you have an idea how to avoid this?

> * Because the entry.js and search.js are Class Components, we cannot use hooks. Is there a plan to change it (and go all into functional components 😍)

Let's discuss this over a beer. ;)

**Comment by sauerben** [comment]
The hook for tracking (like useTracking r.g. ) are only necessary when we are tracking based on lifecycle changes. There is a clever implementation with a dependency array we are using at different Projects.
But there should of cause also be a function for tracking data in a handler like a click you can just call from everywhere. In the end it is just data layer push.

The reason behind the `useTranslation` hook is as far as i know just that we need to know the language. I would be very easy to write a function taking a language and a  token and then translating it. But then we are losing 

The solution to keep the entry.js and the search.js  small is to extract the tracking function. But because we are using lifecycle function to trigger the tracking, the only solution would be using HOC (or Hooks for FC of course)

I am looking forward to the beer 🍸

File: patterns/core/ProductDetailInformation/Buybox.js [hunk 1]
```diff
@@ -45,11 +45,29 @@ export default function Buybox(props) {
   function handleAddToCart() {
     setLoading(true)
 
+    GTM.trackEvent({
+      event: 'add_to_cart',
+      ecommerce: {
+        items: variants.map(prepareTrackingItem),
```
Comment (MrAvantiC) [inline patterns/core/ProductDetailInformation/Buybox.js]: I think this is wrong. We are now tracking as if we would add all available variants to the cart when in reality we are only ever adding one of the products variant to the cart on the detal page.

Also the quantity is missing here

Comment (MrAvantiC) [inline patterns/core/ProductDetailInformation/Buybox.js]: I think this is wrong. We are now tracking as if we would add all available variants to the cart when in reality we are only ever adding one of the products variant to the cart on the detal page.

Also the quantity is missing here.

File: patterns/core/ProductDetailInformation/Buybox.js [hunk 2]
```diff
@@ -45,11 +45,29 @@ export default function Buybox(props) {
...
     setTimeout(() => {
       setLoading(false)
     }, 3000)
   }
 
+  function handleAddToWishlist() {
+    GTM.trackEvent({
+      event: 'add_to_wishlist',
+      ecommerce: {
+        items: variants.map(prepareTrackingItem),
```
Comment (MrAvantiC) [inline patterns/core/ProductDetailInformation/Buybox.js]: Same as for the addToCart event, we are only adding one product/variant to the wishlist, not multiple.

File: patterns/core/ProductList/List.js [hunk 3]
```diff
@@ -14,20 +14,38 @@ export default function List(props) {
...
     <div ref={listRef} className={classes}>
-      {products.map((entry) => {
+      {products.map((entry, index) => {
         if (entry.isBanner) {
           return <Banner key={`banner.${entry.id}`} {...entry} />
         } else {
-          return <ProductTile key={entry.id} {...entry.fields} />
+          return (
+            <ProductTile
+              handleTrackingEvent={() => handleTrackingEvent(entry.id, index)}
```
Comment (MrAvantiC) [inline patterns/core/ProductList/List.js]: Should be index + 1 here for the Non-IT'lers, right? ;)

File: utils/core/tracking/prepareTrackingItem.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
+export default function prepareTrackingItem(product, quantity) {
+  const item = {
+    item_id: product.id || product.articleNumber,
+    item_name: product.title,
+    affiliation: 'makaira',
+    currency: 'EUR',
+    price: parseInt(product.price, 10) || product.price,
+    quantity: quantity || product.quantity || 1,
+  }
+
```
Comment (MrAvantiC) [inline utils/core/tracking/prepareTrackingItem.js:11]: Do we also have brands/manufacturers available?

**Pull request merged**